mini-projects/00-jq-rs/src/input.rs

```diff
@@ -0,0 +1,30 @@
+use anyhow::Result;
+use std::fs::File;
+use std::io::{self, BufRead, BufReader, Lines};
+use std::path::{Path, PathBuf};
+
+pub trait input {
+    fn get_lines(&self) -> Result<Vec<String>>;
+}
+
+pub struct file_input {
+    path: PathBuf,
+}
+
+impl input for file_input {
```

Comment:
nice use of traits, even if you just need one way of implementing for now

Reply:
I see this didn't actually end up getting used..... still nice thought though! curious why you decided to pivot away from this